### DISCUSSION TITLE
Iguana IDE for easy MicroPython programming via Wi-Fi or USB

### DISCUSSION THREAD

**Discussion by robo-start** [micropython/micropython#9333] [Show and tell]
**New IDE for MicroPython**

To program microcontrollers in **MicroPython**, you can use a new programming environment called **Iguana**.

[Iguana MicroPython IDE](https://robostart.ru/iguana/en/)

Iguana supports a wide range of boards: **Pyboard**, **ESP32**, **ESP8266**, **Raspberry Pi Pico**, **Wemos D1**, **Wemos D1 R32**, **Wemos D1 mini**, **ESP-01** and more.

Iguana allows stable access to the device via USB or Wi-Fi, even if the device is running an endless looping algorithm.

Iguana can download any type of file to your device. This can be useful if we want, for example, to add an image to a web server page.

**Why Iguana?**

When experimenting with programming robots via Wi-Fi, you can pay attention to the fact that you have to jump up from the computer quite often in order to stop your robot. The fact is that after the program is interrupted, the robot does not always stop (signals remain on the microcontroller pins), the robot motors continue to work and it always strives to fall down the stairs or break your favorite family vase.

To solve this problem, it is convenient to use the launch of a script that stops the motors after interrupting the current program with the "STOP" button. Iguana allows you to use stop-scripts.

Any program on the device can be marked as a stop-script, which will automatically start after pressing the "STOP" button. Now, when working via Wi-Fi, we can not jump up from the computer every time to stop our robot.

Iguana has a built-in ESP bootloader that allows you to install MicroPython directly from the IDE.

The code editor in Iguana has a large selection of syntax highlighting themes. You can change the font and its size.

**Comment by scruss** [comment]
NB: Iguana IDE is for Windows

**Comment by peterhinch** [comment]
Is it open source?

**Comment by robo-start** [comment]
So far only for Windows, but after the approval of the first users and understanding of the demand for this environment, there will be a version for Linux and Mac. And after code refactoring, an open source version is assumed.

**Comment by teuler** [comment]
Who is behind it? I did not find an Impressum on the website?

**Reply by robo-start** [reply]
The program contains the Impressum if you call "About" in the menu. The website has a feedback form.
